src/document/position.js

```diff
@@ -221,6 +221,25 @@ CKEDITOR.define( [ 'document/rootelement', 'utils', 'ckeditorerror' ], ( RootEle
 		}
 
 		/**
+		 * Returns a new position on the same level as reference position, but with offset shifted left or right.
+		 *
+		 *		 const pos = new Position( [ 1, 2, 8 ], root );
+		 *
+		 *		 Position.createFromPositionAndShift( pos, -3 ) // [ 1, 2, 5 ]
+		 *		 Position.createFromPositionAndShift( pos, 5 )  // [ 1, 2, 13 ]
+		 *
+		 * @param {document.Position} referencePosition Reference position.
+		 * @param {Number} shift Offset shift.
+		 * @returns {document.Position}
+		 */
+		static createFromPositionAndShift( referencePosition, shift ) {
```

Comment:
We may keep it if we want to, but if it's only use is creating "flat" ranges I'd consider removing it.